src/kernel/arch/x86/task.zig

```diff
@@ -0,0 +1,265 @@
+const std = @import("std");
+const expectEqual = std.testing.expectEqual;
+const expectError = std.testing.expectError;
+const assert = std.debug.assert;
+const builtin = @import("builtin");
+const is_test = builtin.is_test;
+const build_options = @import("build_options");
+const mock_path = build_options.arch_mock_path;
+const arch = if (is_test) @import(mock_path ++ "arch_mock.zig") else @import("arch.zig");
+const log = if (is_test) @import(mock_path ++ "log_mock.zig") else @import("../../log.zig");
+const panic = if (is_test) @import(mock_path ++ "panic_mock.zig").panic else @import("../../panic.zig").panic;
+const gdt = @import("gdt.zig");
+const ComptimeBitmap = @import("../../bitmap.zig").ComptimeBitmap;
+const Allocator = std.mem.Allocator;
+
+/// The kernels main stack start as this is used to check for if the task being destroyed is this stack
+/// as we cannot deallocate this.
+extern var KERNEL_STACK_START: *u32;
+
+/// The function type for the entry point.
+const EntryPointFn = fn () void;
+
+/// The bitmap type for the PIDs
+const PidBitmap = if (is_test) ComptimeBitmap(u128) else ComptimeBitmap(u1024);
```

Comment:
It would be good to move the PID stuff to scheduler.zig since all architectures will use them.

Reply:
I might have an idea for moving the task struct out of arch (along with the PID stuff)